### PR TITLE
feat: isValidSignature

### LINKED ISSUE #104: Add `.isValidSignature()` method
### I'm submitting a...

- feature request

### Summary

```js
await myERC725.isValidSignature("Message or hash", signature);
```

If the input parameter is hex + 32 bytes long (char length == 66 AND starts with or without 0x) -> then we assume the input is a hash -> no pre hash before check.
If not, the lib should hash the input first.

Then, it calls `.isValidSignature()` on the contract used at init.
If we get the magic value `0x1626ba7e` -> returns true
If not -> returns false

If call error -> throw error which says "Does the address ${init contract address} support EIP-1271?"

Because the lib can be init with asset or non UP account SC addresses

### Other information

- <https://eips.ethereum.org/EIPS/eip-1271>
- Add documentation accordingly

### PR DESCRIPTION
**Pull request by Hugoo**
Closes #104 

## Docs page

image

image

### CODE REVIEW

**Review by frozeman — APPROVED** [review]

**Comment by frozeman** [comment]
@Hugoo when done?

**Comment by Hugoo** [comment]
will continue working on it by eow - tmrw Fri. 4th

File: src/lib/isValidSignature.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @param {string} messageOrHash
+ * @param {string} signature
+ * @param {string} address the contract address
+ * @returns {Promise<boolean>} Return true if the signature is valid (if the contract returns the magic value), false if not.
+ */
+export const isValidSignature = async (
+  messageOrHash: string,
+  signature: string,
+  address: string,
+  wrappedProvider: EthereumProviderWrapper | Web3ProviderWrapper, // ProviderWrapper
```
Comment (CallumGrindle) [inline src/lib/isValidSignature.ts:44]: Is ProviderWrapper an unused type? Can it be removed?

**Review by CallumGrindle — APPROVED** [review]

File: test/mockProviders.ts [hunk 2]
```diff
@@ -164,14 +188,17 @@ interface EthereumProviderPayload {
 export class EthereumProvider {
   public returnData;
   public supportsInterfaces: string[];
+  public isValidSignature: boolean;
 
   constructor(
     props: { returnData: { key: string; value: string }[] },
     supportsInterfaces: string[],
+    isValidSignature?: boolean, // to mock isValidSignature call. If set to tru, will return magic value.
```
Comment (CallumGrindle) [inline test/mockProviders.ts:196]: typo `true`

Comment (CallumGrindle) [inline test/mockProviders.ts:196]: typo `true`

**Review by frozeman — APPROVED** [review]

**Pull request merged**